test/test_sparse_csr.py

```diff
@@ -1475,15 +1475,19 @@ def run_test_block_addmm_addmv(self, addmv_addmm, c, a, b, op_b=False, op_out=Fa
         out = torch.empty_like(c.mH if op_out and a.shape == b.shape else c)
         addmv_addmm(c, a, b, alpha=alpha, beta=beta, out=out)
 
-        a_bsr = sp.bsr_matrix(
-            (
-                a.values().cpu().numpy(),
-                a.col_indices().cpu().numpy(),
-                a.crow_indices().cpu().numpy(),
-            ),
-            shape=a.shape,
-        )
-        expected = alpha * (a_bsr * b.cpu().resolve_conj().numpy()) + beta * c.cpu().numpy()
+        if dtype not in {torch.half, torch.bfloat16}:
+            a_bsr = sp.bsr_matrix(
+                (
+                    a.values().cpu().numpy(),
+                    a.col_indices().cpu().numpy(),
+                    a.crow_indices().cpu().numpy(),
+                ),
+                shape=a.shape,
+            )
+            expected = alpha * (a_bsr * b.cpu().resolve_conj().numpy()) + beta * c.cpu().numpy()
+        else:
+            expected = alpha * (a.to_dense() @ b) + beta * c
```

Comment:
I see no reason why we can't always compute the NumPy ref in this way. The use of SciPy's bsr just adds an unnecessary SciPy dependency to this test.

Reply:
I agree. But decided not to remove SciPy dep just yet as it is a bit out of scope for this PR.